service/controller/v15/resource/cloudformation/create.go

```diff
@@ -60,6 +60,16 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 			return microerror.Mask(err)
 		}
 
+		enableTerminationProtection := key.EnableTerminationProtection
+		updateTerminationProtection := &cloudformation.UpdateTerminationProtectionInput{
+			EnableTerminationProtection: &enableTerminationProtection,
+			StackName:                   stackInput.StackName,
+		}
+		_, err = r.hostClients.CloudFormation.UpdateTerminationProtection(updateTerminationProtection)
+		if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
Doesn't it have to be disabled before calling delete API?

Reply:
By default is disabled, why?

Reply:
When the cluster is deleted by aws-operator it deletes the CF stack. Will it succeed when termination protection is on?

Reply:
yeah good point, probably yes, e2e in my local still running

Reply:
Is this the only way to define the termination protection? Can we define this via the CF templates? I would expect the default to be set there and having the migration mechanism here only for older clusters. How does the transition work for older clusters? This here is for creation but old clusters will only be updated.

Reply:
> Is this the only way to define the termination protection? Can we define this via the CF templates? I would expect the default to be set there and having the migration mechanism here only for older clusters. How does the transition work for older clusters? This here is for creation but old clusters will only be updated.

I don't find it in the docs. The default is disabled. I have been thought in adding it in the update cycle, but make an extra call only for transition imo is not worthy, but I can do it. I thought to update by hand for the existing clusters

Reply:
> I'm not sure if e2e will catch that. Please check stacks after e2e finishes in the AWS console.

I will test different scenarios before

Reply:
Yeah well, the question is why we want to enable it and how we can enable it. Should all clusters get this? Should only new clusters get this? Is it possible to just enable and disable this setting to any CF stack? With the current implementation only new clusters will get the settings and older ones will not get it because they are already created and the logic of `ApplyCreateChange` will not be executed on updates.

Reply:
I see the benefit of termination protection but doing it via Cloud Formation worries me. As well as the upgrade path there is also the S3 bucket and KMS key that are not managed by CF.

The KMS key is scheduled for deletion in 7 days so we could get it back. The CF protection should prevent downtime as it has most of the critical stuff. But we'd still need manual steps to recover the cluster.

I wonder whether we should have termination protection as part of our API? So its exposed via Happa and gsctl?

Reply:
Well, I am trying to resolve a PM which describes a case of manual stack deletion. One of the ways to make this action harder is protecting the stacks. 

>  Should all clusters get this? 

Yes. 

> Is it possible to just enable and disable this setting to any CF stack? 

Yes

>  With the current implementation only new clusters will get the settings and older ones will not get it because they are already created and the logic of ApplyCreateChange will not be executed on updates.

I know. But my position is to enable this protection manual for existing clusters instead of adding an extra call for every update cycle. It can be argued but it has to be done once and then we dont punish all update cycle with an extra API request

Reply:
Valid points Ross. I feel like this is a candidate of easy change and huge implications. I would like to have this discussed in more depth and having an agreement from the product side here. I am not confident about all this. Sorry to smash that down Fernando. This is not on purpose. :D <3

Reply:
Yes I'd missed that the stack was deleted via the AWS console. But I think having termination protection in our platform could still be useful. We could then propagate it to Cloud Formation.

I think the reason for the change makes total sense. But I'd agree it could have wider implications and it'd be good to get feedback from product.

Reply:
No problem at all. I thought there were no such implications. The idea here was no protecting cluster from deletion (customer side). It was to protect cloud resources for manual actions (like one happened in the PM). In case it could have an impact on the normal flow, then let's have a proper discussion.

Reply:
For me this change LGTM, but I'd like to be reassured stacks are deleted properly on cluster deletion.